src/main/java/com/google/api/codegen/transformer/ImportTypeTransformer.java

```diff
@@ -51,6 +59,39 @@
       fullNames.add(namer.getProtoFileImportFromService(targetInterface));
     }
 
+    return createImportFilesFromNames(fullNames);
+  }
+
+  public List<ImportTypeView> generateProtoFileImports(SurfaceTransformerContext context) {
```

Comment:
Right, BFS traversal is not necessary, the same list is fine. The code would be quite similar to `getServiceImports`.